app/controllers/concerns/publishers/wizardable.rb

```diff
@@ -140,8 +144,9 @@ def include_additional_documents_params(params)
     end
   end
 
-  def completed_steps
-    (vacancy.completed_steps | [current_step.to_s]).compact
+  # Returns an array of completed steps, adding the current step and removing any steps that need to be reset
+  def completed_steps(steps_to_reset: [])
+    (vacancy.completed_steps | [current_step.to_s]).compact - steps_to_reset.map(&:to_s)
```

Comment:
This is the extra feature we added that makes the "contact details submission can cancel the completed confirmation" possible.